ChangeLog.d/clean_pem_buffers.txt

```diff
@@ -0,0 +1,8 @@
+Bugfix
+  * As an optimisation to stop large buffers being written to the stack the
+    raw der data was instead written to the same buffer that the PEM data
+    would later be written to. Unfortunately this would cause data to be left
+    in the buffer after the null termination, which could potentially break
+    the parsing functions if the whole buffer was passed into it.
```

Comment:
That's much better but there's still too much information. Changelog entries are for users' information. It isn't important why the bug arose, but what its consequences where. On the other hand, it might be useful to know when the bug arose (especially in 2.16 branches where this was a regression in 2.16.8 compared to previous 2.16.x). Also the lead lacks context.

Here's how I'd write this changelog entry:

> In PEM writing functions, fill the trailing part of the buffer with null bytes. This guarantees that the corresponding parsing function can read the buffer back, which was the case for mbedtls_x509write_{crt,csr}_pem until this property was inadvertently broken in Mbed TLS 2.19.0. Fix #3682.